### PR TITLE
Add solution for 65. Valid Number

### PR DESCRIPTION
**Pull request by yiru941013**
@kuangami

### CODE REVIEW

File: java/65_Valid_Number.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            }
+            else if (s.charAt(i) == 'e') {
+                if (e || !number)
+                    return false;
+                e = true;
+                numberAfterE = false;
+            }
+            else if (s.charAt(i) == '.') {
+                if (e || dot)
+        	        return false;
```
Comment (kuangami) [inline java/65_Valid_Number.java]: tab

File: java/65_Valid_Number.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                dot = true;
+            }
+            else 
+                return false;
+
```
Comment (kuangami) [inline java/65_Valid_Number.java]: two tabs

File: java/65_Valid_Number.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+            else 
+                return false;
+        		
+        }
+        return number && numberAfterE;
+    }
+    
+    public static void main(String[] args) {
+        String s1 = "1", s2 = " +1", s3 = "2s2e342",
+    	    s4 = " +-1 ", s5 = "2.4", s6 = ".4", s7 = "2e4", s8 = "-12e-10";
```
Comment (kuangami) [inline java/65_Valid_Number.java]: tab
To solve this tab issue, go to your eclipse preferences: General -> Editors -> Text Editors. Check "Insert spaces for tabs". Then you are free to use tabs!

**Review by kuangami — CHANGES_REQUESTED** [review]
Smart way to use two flags before and after 'e' for result output!

File: java/65_Valid_Number.java (hunk 3 above)
Comment (yiru941013) [reply java/65_Valid_Number.java]: 👍

Comment (kuangami) [reply java/65_Valid_Number.java]: Click **Window** -> **Preferences**
Expand **Java** -> **Code Style**
Click **Formatter**
Click the **Edit** button
Click the **Indentation** tab
Under **General Settings**, set **Tab policy** to: Spaces only
Save the profile with a new name
Click **OK** to apply the changes.

**Review by kuangami — APPROVED** [review]

**Pull request merged**